client/python/gradio_client/client.py

```diff
@@ -603,6 +605,137 @@ def _get_config(self) -> dict:
+            private: Whether the space hosting the discord bot is private. The visibility of the discord bot itself is set via the discord website. See https://huggingface.co/spaces/freddyaboulton/test-discord-bot-v1
+            persist_state: Currently has no effect.
+        """
+        if self.config["mode"] != "chat_interface":
+            raise ValueError(
+                "Only demos built with gr.ChatInterface can be deployed as discord bots automatically for now."
+            )
+
+        if not command_names:
+            command_names = ["chat"]
+
+        assert (
+            len(command_names) == 1
+        ), f"gr.ChatInterface can only accept one command name. Received {', '.join(command_names)}"
+        api_names = ["chat"]
+
+        is_private = False
+        if self.space_id:
+            is_private = huggingface_hub.space_info(self.space_id).private
+            if is_private:
+                assert hf_token, (
+                    f"Since {self.space_id} is private, you must explicitly pass in hf_token "
+                    "so that it can be added as a secret in the discord bot space."
+                )
+
+        if to_id:
+            if "/" in to_id:
+                to_id = to_id.split("/")[1]
+            space_id = huggingface_hub.get_full_repo_name(to_id, token=hf_token)
+        else:
+            if self.space_id:
+                space_id = f'{self.space_id.split("/")[1]}-gradio-discord-bot'
+            else:
+                space_id = f"gradio-discord-bot-{secrets.token_hex(4)}"
+            space_id = huggingface_hub.get_full_repo_name(space_id, token=hf_token)
+
+        api = huggingface_hub.HfApi()
+
+        try:
+            huggingface_hub.space_info(space_id)
+            first_upload = False
+        except huggingface_hub.utils.RepositoryNotFoundError:
+            first_upload = True
+
+        huggingface_hub.create_repo(
+            space_id,
+            repo_type="space",
+            space_sdk="gradio",
+            token=hf_token,
+            exist_ok=True,
+            private=private,
+        )
+
+        with open(str(Path(__file__).parent / "templates" / "discord_chat.py")) as f:
+            app = f.read()
+        app = app.replace("<<app-src>>", self.src)
+        app = app.replace("<<command-name>>", api_names[0])
+
+        with tempfile.NamedTemporaryFile(mode="w", delete=False) as app_file:
+            with tempfile.NamedTemporaryFile(mode="w", delete=False) as requirements:
+                app_file.write(app)
+                requirements.write("\n".join(["discord.py==2.3.1", "apscheduler"]))
+        readme = None
+        if first_upload:
+            with tempfile.NamedTemporaryFile(mode="w", delete=False) as readme:
+                content = f"""
```

Comment:
Maybe instead we can do this in a temporary directory and then use `upload_folder` to upload all its files, then you have no need to specify each commit operation